AsyncDisplayKit/ASNetworkImageNode.mm

```diff
@@ -377,7 +396,8 @@ - (void)_lazilyLoadImageIfNecessary
 - (void)asyncdisplaykit_asyncTransactionContainerStateDidChange
 {
   if (self.asyncdisplaykit_asyncTransactionContainerState == ASAsyncTransactionContainerStateNoTransactions) {
-    if (self.layer.contents != nil && [self.delegate respondsToSelector:@selector(imageNodeDidFinishDecoding:)]) {
+    ASDN::MutexLocker l(_lock);
+    if (self.layer.contents != nil && _delegateSupportsImageNodeDidFinishDecoding) {
       [self.delegate imageNodeDidFinishDecoding:self];
```

Comment:
Should we avoid holding the lock during the delegate calls?  I'm not totally sure, this is probably OK since they are recursive locks, just wondering if any use case might have some other thread interdependency.